sklearn/metrics/tests/test_common.py

```diff
@@ -691,29 +691,33 @@ def test_thresholded_invariance_string_vs_numbers_labels(name):
             assert_raises(ValueError, metric, y1_str.astype('O'), y2)
 
 
-invalids = [([0, 1], [np.inf, np.inf]),
-            ([0, 1], [np.nan, np.nan]),
-            ([0, 1], [np.nan, np.inf])]
+invalid_nan_inf = [([0, 1], [np.inf, np.inf]),
+                   ([0, 1], [np.nan, np.nan]),
+                   ([0, 1], [np.nan, np.inf])]
 
 
 @pytest.mark.parametrize(
         'metric',
         chain(THRESHOLDED_METRICS.values(), REGRESSION_METRICS.values()))
 def test_regression_thresholded_inf_nan_input(metric):
 
-    for y_true, y_score in invalids:
+    for y_true, y_score in invalid_nan_inf:
         assert_raise_message(ValueError,
-                             "contains NaN, infinity",
+                             "Input contains NaN, infinity or a value too"
+                             " large for dtype('float64').",
                              metric, y_true, y_score)
 
 
+invalid_bin_cont = [(['a', 'b', 'a'], [0.1, 0.2, 0.3])]
+
+
 @pytest.mark.parametrize('metric', CLASSIFICATION_METRICS.values())
-def test_classification_inf_nan_input(metric):
+def test_classification_binary_continuous_input(metric):
```

Comment:
I think we would still need a test like this for classification metrics (above is for regression).